docs/advanced/tls_porting.md

```diff
@@ -24,46 +24,56 @@ mbed TLS distinguishes between strong and weak entropy sources. Of the sources r
 
 The preferred way to provide a custom entropy source:
 
-1. Implement the `mbedtls_hardware_poll` function to let it access the device's entropy source.
-2. Set the appropriate macros for your target in `hal/targets.json`.
+1. Implement the `trng_api.h` to let mbed TLS access the device's entropy source.
+2. Indicate in `hal/targets.json` that your target has an entropy source.
 
 The next two sections explain how to do this.
 
-## How to implement the hardware poll
+## How to implement the TRNG api
 
-The `mbedtls_hardware_poll` function is declared in `entropy_poll.h`:
+The implementation of this interface needs to be in the HAL.
+
+### Data structure
+
+You have to define a structure `trng_s` that holds all the information needed to operate the peripherial and describe its state.
+
+### Initialisation and deinitialisation
+
+Provide means for initialising and deinitialising the peripherial with the following functions:
```

Comment:
Isn't that a fragment? Don't you need to say something like 'You need to provide means....'.